### PR TITLE
SELV3-626: Cascade delete fixes

### PR DESCRIPTION
**Pull request by anawrotsoldevelo**

### CODE REVIEW

File: src/main/java/org/openlmis/integration/dhis2/domain/element/DataElement.java [hunk 1]
```diff
@@ -15,62 +15,67 @@
...
+import org.openlmis.integration.dhis2.domain.schedule.Schedule;
 
 @Entity
 @TypeName("DataElement")
 @Table(name = "data_element", schema = "dhis2")
+@Getter
+@Setter
 @NoArgsConstructor
-@AllArgsConstructor
+@RequiredArgsConstructor
```
Comment (NikolaLaskowska) [inline src/main/java/org/openlmis/integration/dhis2/domain/element/DataElement.java:46]: Isn't it inconsistent with @NoArgsConstructor annotation?

Comment (anawrotsoldevelo) [reply src/main/java/org/openlmis/integration/dhis2/domain/element/DataElement.java:46]: Annotation @NoArgsConstructor is being used there to create new Entity from DTO object.
Annotation @RequiredArgsConstructor is being used in normal aplication flow and for testing purposes.
I think both are required here.

Comment (sradziszewski) [reply src/main/java/org/openlmis/integration/dhis2/domain/element/DataElement.java:46]: If you need an argument constructor in your entity, maybe consider using the **AllArgsConstructor** annotation, like in [FacilityType](https://github.com/OpenLMIS/openlmis-referencedata/blob/master/src/main/java/org/openlmis/referencedata/domain/FacilityType.java) or [User](https://github.com/OpenLMIS/openlmis-referencedata/blob/master/src/main/java/org/openlmis/referencedata/domain/User.java)

Comment (anawrotsoldevelo) [reply src/main/java/org/openlmis/integration/dhis2/domain/element/DataElement.java:46]: DataElement has bidirectional relationship with Schedule. In it's class there is a reference to list of all schedules connected with this element. We don't want this list to be set from the constructor, so AllArgsConstructor is not applicable here.

**Review by sradziszewski — APPROVED** [review]

**Pull request merged**